source/ProtonPack/ProtonPack.ino

```diff
@@ -4663,13 +4683,79 @@ void readEEPROM() {
 
     // Update the LED counts for the Proton Pack.
     updateProtonPackLEDCounts();
+
+    // Read our configration object from the EEPROM.
+    objConfigEEPROM obj_config_eeprom;
+    unsigned int i_eepromConfigAddress = EEPROM.length() / 2;
+
+    EEPROM.get(i_eepromConfigAddress, obj_config_eeprom);
+
+    if(obj_config_eeprom.stream_effects > 0 && obj_config_eeprom.stream_effects != 255) {
+      if(obj_config_eeprom.stream_effects > 1) {
+        b_stream_effects = true;
+      }
+      else {
+        b_stream_effects = false;
+      }
+    }
+
+    if(obj_config_eeprom.stream_effects > 0 && obj_config_eeprom.stream_effects != 255) {
+      if(obj_config_eeprom.stream_effects > 1) {
+        b_cyclotron_single_led = true;
+      }
+      else {
+        b_cyclotron_single_led = false;
+      }
+    }
+  }
+}
+
+void clearConfigEEPROM() {
+  // Clear out the EEPROM data for the configuration settings only.
+  unsigned int i_eepromConfigAddress = EEPROM.length() / 2;
+
+  for(unsigned int i = 0 ; i < sizeof(objConfigEEPROM); i++) {
+    EEPROM.put(i_eepromConfigAddress, 0);
+
+    i_eepromConfigAddress++;
   }
+
+  updateCRCEEPROM();
+}
+
+void saveConfigEEPROM() {
+  // Proton Stream Impact Effects
+  // Three LED setting
+
+  uint8_t i_single_led = 2;
+  uint8_t i_proton_stream_effects = 2;
+
+  if(b_cyclotron_single_led != true) {
+    i_single_led = 1;
+  }
+
+  if(b_stream_effects != true) {
+    i_proton_stream_effects = 1;
+  }
+
+  unsigned int i_eepromConfigAddress = EEPROM.length() / 2;
```

Comment:
Is this just looking for the midpoint of the available EEPROM space and getting the starting byte of memory to use? Do we know this would always be an even address to avoid an error? What if there are more EEPROM settings later, how might you select the starting point? It feels like it may be more safe to carve out more distinct values, leaving buffer in your first space for the LED values to avoid potential overlap, rather than just an arbitrary point in the memory space.

Reply:
This is the midway point in the entire EEPROM buffer. I figured the first half of the buffer for the LED stuff and start at the midpoint for the configuration settings. There is so much buffer space, we are barely using any of it. Lots of space available.

Reply:
I see, they say there's 1024 bytes available, which for this purpose is indeed plenty. I'm just wondering if there could be more features you may add later that may somehow require a larger space, and now there's less than 512 bytes of contiguous space available anywhere. If you reserved 64 bytes for the LED and 64 for the config, you've still got 896 bytes available if you need to store something bigger.

Reply:
Actually there is 4096 bytes on the chips being used in the Proton Pack and Neutrona Wand :)

Reply:
Well then, I read the wrong spec sheet!